tests/test_functional.py

```diff
@@ -0,0 +1,48 @@
+import logging
+import os
+import socket
+from unittest import mock, TestCase, skipUnless
+from urllib.parse import urlparse
+
+from mtp_transaction_uploader import upload, settings
+
+logger = logging.getLogger('mtp')
+
+
+@skipUnless('RUN_FUNCTIONAL_TESTS' in os.environ, 'functional tests are disabled')
+class FileUploadFunctionalTestCase(TestCase):
+
+    def setUp(self):
+        self.load_test_data()
```

Comment:
to not duplicate the test data reloading method, could this be `FunctionalTestCase.load_test_data(None)`? not too bothered though since of course this class cannot extend that one.